netlogo-gui/src/main/app/codetab/IncludesMenu.scala

```diff
@@ -0,0 +1,107 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.app.codetab
+
+import java.awt.{ Dimension, FileDialog }
+import java.awt.event.{ ActionEvent, ActionListener }
+import java.io.File
+import java.util.{ ArrayList => JArrayList, Collections }
+import javax.swing.{ AbstractAction, JMenuItem, JOptionPane, JPopupMenu }
+
+import scala.collection.JavaConverters._
+
+import org.nlogo.api.Exceptions
+import org.nlogo.app.common.TabsInterface
+import org.nlogo.awt.UserCancelException
+import org.nlogo.swing.{ FileDialog => SwingFileDialog, ToolBarMenu }
+import org.nlogo.window.{ Events => WindowEvents }
+
+class IncludesMenu(target: CodeTab, tabs: TabsInterface)
+extends ToolBarMenu("Includes")
+with WindowEvents.CompiledEvent.Handler
+{
+
+  // If we're empty, we have no size, are invisible and don't affect our parent's layout
+  private var isEmpty = true
+
+  private var includesTable: Map[String, String] = null
+
+  def handle(e: WindowEvents.CompiledEvent) {
+    if(e.sourceOwner.isInstanceOf[CodeTab])
+      updateVisibility()
+  }
+
+  def updateVisibility() {
+    isEmpty = target.getIncludesTable.isEmpty
+    revalidate()
+    super.doLayout()
+  }
+
+  override def populate(menu: JPopupMenu) {
+    target.getIncludesTable match {
+      case Some(includesTable) =>
+        this.includesTable = includesTable
+        val includes = new JArrayList[String]
+        includesTable.keys.foreach(includes.add)
+        Collections.sort(includes, String.CASE_INSENSITIVE_ORDER)
+        for(include <- includes.asScala if include.endsWith(".nls") && include.size > 4)
```

Comment:
We shouldn't be using a `JArrayList` here. Let's use `includesTable.keys.toSeq.sortBy(_.toUpperCase)`.